#4 lesson 7 routing component creation

Sapienti17 opened the pull request.

Sapienti17 committed:
added edits

Sapienti17 committed:
Task-2_completed

Sapienti17 committed:
fix a past mistake

Sapienti17 committed:
fixed bugs

Sapienti17 committed:
added new features

Sapienti17 committed:
completed task of lesson 7

pocahontassss commented on src/app/users-list/users-list.component.ts:
```diff
@@ -0,0 +1,56 @@
+import { Component, inject } from '@angular/core';
+import { NgFor, NgForOf } from '@angular/common';
+
+interface IUser {
+  'id': number,
+  'name': string,
+  'username': string,
+  'email': string,
+  'address': {
+    'street': string,
+    'suite': string,
+    'city': string,
+    'zipcode': string,
+    'geo': {
+      'lat': string,
+      'lng': string
+    }
+  }
+  'phone': string,
+  'website': string,
+  'company': {
+    'name': string,
+    'catchPhrase': string,
+    'bs': string
+  }
+}
```
Свойства интерфейса не должны быть в кавычках

pocahontassss commented on src/app/users-list/users-list.component.ts:
```diff
@@ -0,0 +1,56 @@
+
+  constructor() {
+    this.apiService.get<IUser[]>('https://jsonplaceholder.typicode.com/users').subscribe(
+      response => {
```
типизировать responce

pocahontassss commented on src/app/users-list/users-list.component.ts:
```diff
@@ -0,0 +1,56 @@
+  }
+
+  deleteUser(id: number) {
+    this.users = this.users.filter(user => user.id !== id);
```
типизировать user

pocahontassss commented on src/app/users-list/users-list.component.html:
```diff
@@ -0,0 +1,11 @@
+<div class="user-list__container">
+  <div class='user-list__items' *ngFor="let item of users">
+    <p>{{ item.name }}</p>
+    <p>{{ item.email }}</p>
+    <p>{{ item.address.city }}</p>
+    <p>{{ item.phone }}</p>
+    <p>{{ item.website }}</p>
+    <p>{{ item.company.name }}</p>
+    <button class="user-list__btn" (click)="deleteUser(item.id)">Удалить</button>
```
let user of users (item - абстракция)

Sapienti17 committed:
fix mistakes

Sapienti17 committed:
Update users-list.component.html

исправил имя класса, которое случайно переименовал при групповом переименовании item на user
